src/app/spectatorDashboard/Postwinners/page.tsx

```diff
@@ -1,31 +1,135 @@
+  const roomID = getItemFromLS('roomId');
+  const roomUuid = getItemFromLS('roomUuid');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [roomUsers, setRoomUsers] = useState<null | []>(null);
+  const [winnnerTeamData, setWinnnerTeamData] = useState<null | GameRoomType>(null);
+  const [formData, setFormData] = useState<winnerFormType | object>({});
+  const [isEdit, setIsEdit] = useState<boolean>(false);
+
+  console.log('checking values before submit', formData);
+
+  const getAllTeams = async () => {
+    try {
+      const roomTeamsGet = await sendRequest(`/room/getTeam/${roomID}`);
+      setRoomUsers(roomTeamsGet?.data?.teams);
+    } catch (error) {
+      console.log('Error in get team data', error);
+    }
+  };
+
+  const getWinningTeams = async () => {
+    try {
+      const winnerTeamsResult = await sendRequest(`/winners/get-players/${roomUuid}`);
+      setWinnnerTeamData(winnerTeamsResult?.data);
+      if (roomUsers?.length === 0) {
+        toast.error('No team data found !');
+      }
+    } catch (error) {
+      console.log('Error in get team data', error);
+    }
+  };
 
-  console.log('isLoading', isLoading);
-  const columns: string[] = [
-    'Team Name',
-    'Chicken Dinner',
-    'Highest Kill',
-    'Second Winner',
-    'Third Winner',
-  ];
+  const getWinnerPostData = (index: number, teamName: string) => {
+    if (winnnerTeamData) {
+      return winnnerTeamData.teams?.[index]?.teamName === teamName
+        ? winnnerTeamData.teams?.[index]?.prizeTitles
+        : '';
+    }
+  };
 
-  const handleLoading = () => {
+  const submitWinningTeam = async () => {
     setIsLoading(true);
-    setTimeout(() => {
+    try {
+      const response = await sendRequest(`/winners/players/${roomUuid}`, {
+        method: `${winnnerTeamData ? 'PUT' : 'POST'}`,
+        data: Object.values(formData),
+      });
+      toast.success('Winning team update successfully');
+      router.push('/spectatorDashboard');
+      if (response.status !== 200) {
+        toast.error('Fail to update Winning team data !');
+      }
+      setIsLoading(false);
+    } catch (error) {
       setIsLoading(false);
-    }, 1000);
+    }
+  };
+
+  useEffect(() => {
+    getAllTeams();
+    getWinningTeams();
+  }, []);
+
+  useEffect(() => {
+    (roomUsers || []).forEach(({ teamName }) => {
+      setFormData((prev) => ({
+        ...prev,
+        [teamName]: {
+          //@ts-ignore
+          ...prev[teamName],
+          teamName: teamName,
+          chickenDinner: 0,
+          highestKill: 0,
+          firstWinner: 0,
+          secondWinner: 0,
+        },
+      }));
+    });
+  }, [roomUsers]);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>, teamName: string) => {
+    setIsEdit(true);
+    const { value, name } = e.target;
+    const point = Number(value);
+    roomUsers?.forEach(({ teamName }) => {
+      setFormData((prev) => ({
+        ...prev,
+        [teamName]: {
+          //@ts-ignore
```

Comment:
why ignore?

Reply:
There is some Unexcpected error that I'm not able to resolve and If I made some change then logic will no work here so it's batter to ignore type for this case only, as per my understanding